sycl/include/CL/sycl/intel/group_algorithm.hpp

```diff
@@ -40,11 +40,11 @@ template <typename Group>
 typename Group::linear_id_type get_local_linear_id(Group g);
 
 #ifdef __SYCL_DEVICE_ONLY__
-#define __SYCL_GROUP_GET_LOCAL_LINEAR_ID(D)                                    \
-  template <>                                                                  \
-  group<D>::linear_id_type get_local_linear_id<group<D>>(group<D>) {           \
-    nd_item<D> it = cl::sycl::detail::Builder::getNDItem<D>();                 \
-    return it.get_local_linear_id();                                           \
+#define __SYCL_GROUP_GET_LOCAL_LINEAR_ID(D)                                                   \
+  template <>                                                                                 \
+  group<D>::linear_id_type get_local_linear_id<group<D>>(group<D>) {                          \
+    nd_item<D> it = cl::sycl::detail::Builder::getElement(static_cast<nd_item<D>*>(nullptr)); \
```

Comment:
This is a lot less readable than what we had before, and it's not clear to me why passing a `nullptr` into this function is necessary or what it means.

Is there some reason not to support the old `getNDItem<D>()`-style interfaces as well as your new `getElement()` interface?  Even if `getNDItem<D>()` expanded to exactly what's written here, I think that would improve readability significantly.

Reply:
It's kind of metaprogramming stuff. In other place of the code I use it to choose the right overload at compile-time. It allows me to get rid of several unnecessary overloads and aliases in the code.
However, in this particular place it doesn't help somehow. I need to use it because in current commit I have removed getNDItem.
Although, I expect the readability comment in this place. And basically I agree with that. I also thought about leaving getNDItem as is with just redirection to the statement with static_cast.  I may just return back getNDItem without any problem here. It adds one more interface but it also improves the readability.